The analyses directory consists primarily of files generated by Nanopolish and the helper script calculate_methylation_frequency.py (Simpson et al. 2017.). Files with the suffix `thresh.tsv` or `thresh10.tsv` are tab-delimited files summarizing methylation frequency for each sample at genomic coordinates. They were originally generated by calculate_methylation_frequency.py, then further filtered by minimum read count, which was 3 reads in the case of `thresh.tsv` files and 10 reads in the case of `thresh10.tsv`. `thresh.tsv` files are alignments of each symbiotic and aposymbiotic anemone to the draft genome, while the `thresh10.tsv` file reflects the initial analysis of a single aposymbiotic specimen to the transcriptome from [Kitchen et al. 2015](https://datadryad.org/stash/dataset/doi:10.5061/dryad.3f08f).

Sample info

bar01 = anemone A4 (aposymbiotic)
bar02 = anemone G2 (symbiotic)
bar03 = anemone A3 (aposymbiotic)
bar04 = anemone G4 (symbiotic)
bar05 = anemone A1 (aposymbiotic)
bar06 = anemone G3 (symbiotic)

